### PR TITLE
6071 - ET - server inactive session timeout

### PR DESCRIPTION
**Pull request by etelmon**
- Fix issue - Token expiration is no longer supported since migration to jsonwebtoken 5x.

> Make sure that when the user closes without logging out the browser or all tabs connected via the socket to the zerv server or the network is lost for a duration, the session is invalidated after some time.
> The time is based on the inactive session timeout.

- Converted all remaining mocha unit tests to jasmine
- Jwt token expires based on the maximum ACTIVE session time out (iat and exp values)

**Run**
`npm test`

### CODE REVIEW

File: test/utils.spec.js [hunk 1]
```diff
@@ -115,11 +114,10 @@ describe('utils', () => {
             expect(timeout.remainingMins).toEqual(0);
             jasmine.clock().tick(39 * 24 * 60 * 60 * 1000);
             utils.clearLongTimeout(timeout);
-            expect(fn).not.toHaveBeenCalled();           
-            expect(utils._setTimeout).toHaveBeenCalledTimes(2);       
+            expect(fn).not.toHaveBeenCalled();
+            expect(utils._setTimeout).toHaveBeenCalledTimes(2);
```
Comment (Jacob-Friesen) [inline test/utils.spec.js:118]: `zerv-ng-core` review: https://github.com/z-open/zerv-ng-core/pull/18#pullrequestreview-517228915

File: package.json [hunk 2]
```diff
@@ -19,17 +19,15 @@
...
+    "test": "./node_modules/jasmine/bin/jasmine.js test/*.spec.js",
     "eslint": "./node_modules/eslint/bin/eslint.js lib/**/*.js test/**/*.js",
     "eslintfix": "./node_modules/eslint/bin/eslint.js --fix lib/**/*.js test/**/*.js"
   },
   "license": "MIT",
   "dependencies": {
     "core-transaction": "git://github.com/z-open/core-transaction#0.0.4",
     "ioredis": "4.14.1",
-    "jsonwebtoken": "5.0.0",
+    "jsonwebtoken": "8.5.1",
```
Comment (Jacob-Friesen) [inline package.json:30]: Lodash and eslint should be upgraded too to get rid of some security vulnerabilities.

File: test/utils.spec.js [hunk 3]
```diff
@@ -115,11 +114,10 @@ describe('utils', () => {
...
             expect(utils._setTimeout.calls.argsFor(0)).toEqual([jasmine.any(Function), 1728000000]);
             expect(utils._setTimeout.calls.argsFor(1)).toEqual([jasmine.any(Function), 1728000000]);
         });
     });
-
```
Comment (Jacob-Friesen) [inline test/utils.spec.js:124]: ` npm run eslint` gives this error:
```
zerv-core/test/cache.service.spec.js
  123:56  error  Missing semicolon  semi
```

**Review by Jacob-Friesen — CHANGES_REQUESTED** [review]
Everything seems fine apart from some failing tests.

File: test/utils.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (etelmon) [reply test/utils.spec.js:124]: gulp should not work. it is npm test.

File: package.json (hunk 2 above)
Comment (etelmon) [reply package.json:30]: Updated lodash for now. eslint is very recent.

File: test/utils.spec.js (hunk 3 above)
Comment (etelmon) [reply test/utils.spec.js:124]: Fixed lint error.

[the remark this answers is not in the document]
Comment (Jacob-Friesen) [reply test/utils.spec.js:124]: It is now good.

**Review by Jacob-Friesen — APPROVED** [review]

**Pull request merged**